src/features/wallet/scan/ScanView.tsx

```diff
@@ -0,0 +1,182 @@
+import { StyleSheet } from 'react-native'
+import { BarCodeScanner, BarCodeScannerResult } from 'expo-barcode-scanner'
+import BottomSheet from 'react-native-holy-sheet'
+import { useSafeAreaInsets } from 'react-native-safe-area-context'
+import { Address } from '@helium/crypto-react-native'
+import Box from '../../../components/Box'
+import Text from '../../../components/Text'
+import Crosshair from './Crosshair'
+import { wp } from '../../../utils/layout'
+import Close from '../../../assets/images/close.svg'
+import TouchableOpacityBox from '../../../components/TouchableOpacityBox'
+import { triggerNavHaptic, triggerNotification } from '../../../utils/haptic'
+import sleep from '../../../utils/sleep'
+
+type Props = {
+  fromSend: boolean
+}
+
+const ScanView = ({ fromSend = false }: Props) => {
+  const [scanned, setScanned] = useState<boolean>(false)
+  const navigation = useNavigation()
+
+  const { result: permissions } = useAsync(
+    BarCodeScanner.requestPermissionsAsync,
+    [],
+  )
+
+  const navBack = () => {
+    navigation.goBack()
+    triggerNavHaptic()
+  }
+
+  const handleBarCodeScanned = async ({ data }: BarCodeScannerResult) => {
+    if (scanned) return
+
+    // TODO validate other types of qr codes
+    if (!Address.isValid(data)) {
+      setScanned(true)
+      setTimeout(() => setScanned(false), 1000)
+      triggerNotification('error')
+      return
+    }
+
+    const scanResult = {
+      type: 'payment',
+      address: data,
+    }
+
+    setScanned(true)
+    triggerNotification('success')
+
+    if (fromSend) {
+      navigation.navigate('Send', { scanResult })
+    } else {
+      navigation.navigate('Wallet')
+      await sleep(200)
+      // TODO this doesn't work here to pass the params
+      // might need to use redux or restructure the nav stack
+      navigation.navigate('Send', { scanResult })
+    }
+  }
+
+  if (!permissions) {
+    return <Box flex={1} backgroundColor="black" />
+  }
+
+  if (!permissions.granted) {
+    return (
+      <Box flex={1} backgroundColor="black">
+        <CloseButton safeOffset={!fromSend} onPress={navBack} />
+        <Box flex={1} justifyContent="center" alignItems="center">
+          <Text color="white" textAlign="center">
+            No access to camera
+          </Text>
+        </Box>
+      </Box>
+    )
+  }
+
+  return (
+    <Box flex={1}>
+      <BarCodeScanner
+        onBarCodeScanned={handleBarCodeScanned}
+        barCodeTypes={[BarCodeScanner.Constants.BarCodeType.qr]}
+        style={StyleSheet.absoluteFillObject}
+      />
+      <CloseButton safeOffset={!fromSend} onPress={navBack} />
+      <Box flex={0.7} justifyContent="center" alignItems="center">
+        <Crosshair width={wp(65)} height={wp(65)} color="white" />
+      </Box>
+      <BottomSheet snapPoints={[300, 500]} initialSnapIndex={0}>
+        <Box>
+          <Text color="black" fontSize={18} fontWeight="600" marginBottom="s">
+            Find a QR Code to scan
```

Comment:
this section is just a random idea I had. if we keep it, we can move these to translations/variants etc